File: StykerDG.FarmForge.DataModel/Models/Location.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace StrykerDG.FarmForge.DataModel.Models
{
    [Table("Location")]
    public class Location : BaseModel
    {
        [Key]
        public int LocationId { get; set; }
        public string Name { get; set; }
        public string Label { get; set; }
    }
}

File: StrykerDG.FarmForge.Client/farmforge_client/lib/models/general/location.dart
class Location {
  int locationId;
  int parentLocationId;
  String name;
  String label;

  Location({
    this.locationId,
    this.parentLocationId,
    this.name,
    this.label
  });

  factory Location.fromMap(Map<String, dynamic> data) {
    return Location(
      locationId: data['LocationId'],
      parentLocationId: data['ParentLocationId'],
      name: data['Name'],
      label: data['Label']
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'LocationId': this.locationId,
      'ParentLocationId': this.parentLocationId,
      'Name': this.name,
      'Label': this.label
    };
  }
}